experimental/native_file_system/virtual_root_provider_win.cc

```diff
@@ -1,12 +1,45 @@
-// Copyright (c) 2014 Intel Corporation. All rights reserved.
-// Use of this source code is governed by a BSD-style license that can be
-// found in the LICENSE file.
-
-#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
-
-#include "base/logging.h"
-
-VirtualRootProvider::VirtualRootProvider() {
-  // TODO(darktears): Windows support to be added.
-  NOTIMPLEMENTED();
-}
+// Copyright (c) 2014 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "xwalk/experimental/native_file_system/virtual_root_provider.h"
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+
+bool VirtualRootProvider::testing_enabled_ = false;
+
+VirtualRootProvider::VirtualRootProvider() {
+  if (!virtual_root_map_.empty())
+    return;
+
+  if (testing_enabled_) {
+    base::GetTempDir(&home_path_);
+  } else {
+    home_path_ = base::GetHomeDir();
+  }
+
+  virtual_root_map_["DESKTOP"] =
+      home_path_.Append(FILE_PATH_LITERAL("Desktop"));
+  virtual_root_map_["DOWNLOADS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Downloads"));
+  virtual_root_map_["DOCUMENTS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Documents"));
+  virtual_root_map_["MUSIC"] =
+      home_path_.Append(FILE_PATH_LITERAL("Music"));
+  virtual_root_map_["PICTURES"] =
+      home_path_.Append(FILE_PATH_LITERAL("Pictures"));
+  virtual_root_map_["VIDEOS"] =
+      home_path_.Append(FILE_PATH_LITERAL("Videos"));
```

Comment:
In windows such folders are called `Known Folder`, https://msdn.microsoft.com/en-us/library/windows/desktop/dd378457(v=vs.85).aspx.
These folders can be redirected to other paths than just under home directory by API, `SHSetKnownFolderPath`,  https://msdn.microsoft.com/en-us/library/windows/desktop/bb762249(v=vs.85).aspx.
So the better way for Windows is `SHGetKnownFolderPath`, https://msdn.microsoft.com/en-us/library/windows/desktop/bb762188(v=vs.85).aspx.